backend/src/packages/articles/article.service.ts

```diff
@@ -105,6 +111,49 @@ class ArticleService implements IService {
     return article.toObject();
   }
 
+  public async getUserActivity(
+    userId: number,
+  ): Promise<UserActivityResponseDto> {
+    const currentDate = new Date();
+    const sixMonthAgo = subMonths(currentDate, 6);
```

Comment:
Check **B31**

Reply:
I didn't quiet understand you, what exactly here is a 'magic value'? There are constants everywhere, or do you mean to move these to the constants file?

Reply:
Or do you mean to move these value to the constants?
sixMonthAgo.getFullYear(),
sixMonthAgo.getMonth(),
sixMonthAgo.getDate() + index,

Reply:
Should I also create the constant here: `for (let index = 0; index <= daysInHalfYear; index++) {`?